Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -0,0 +1,351 @@
+    static let historySize = 10
+    static let TopSiteSingleRowHeight: CGFloat = 120
+    static let TopSiteDoubleRowHeight: CGFloat = 220
+}
+
+class ActivityStreamPanel: UIViewController {
+    weak var homePanelDelegate: HomePanelDelegate? = nil
+    private let profile: Profile
+
+    lazy private var tableView: UITableView = {
+        let tableView = UITableView(frame: CGRect.zero, style: .Grouped)
+        tableView.registerClass(SimpleHighlightCell.self, forCellReuseIdentifier: "Cell")
+        tableView.registerClass(ASHorizontalScrollCell.self, forCellReuseIdentifier: "TopSite")
+        tableView.registerClass(HighlightCell.self, forCellReuseIdentifier: "Highlight")
+        tableView.backgroundColor = ASPanelUX.backgroundColor
+        tableView.separatorStyle = .None
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.rowHeight = UITableViewAutomaticDimension
+        tableView.separatorInset = UIEdgeInsetsZero
+        tableView.estimatedRowHeight = 65
+        tableView.estimatedSectionHeaderHeight = 15
+        tableView.sectionHeaderHeight = UITableViewAutomaticDimension
+        return tableView
+    }()
+
+    private let topSiteHandler = ASHorizontalScrollSource()
+
+    var topSites: [TopSiteItem] = []
+    var history: [Site] = []
+
+    init(profile: Profile) {
+        self.profile = profile
+        super.init(nibName: nil, bundle: nil)
+
+        self.profile.history.setTopSitesCacheSize(Int32(ASPanelUX.topSitesCacheSize))
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationFirefoxAccountChanged, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationProfileDidFinishSyncing, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationFirefoxAccountChanged, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationProfileDidFinishSyncing, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        reloadTopSites()
+
+        reloadRecentHistory()
```

Comment:
Also, even though the `reload*` methods are async, I would feel more comfortable having the adding/setting up of the table view happen before we call reload since they require the table view to be there. Since `tableView` is also lazy, this means that initialization might happen while reloading instead of the `view.addSubview` call below.